restaurants/filters.py

```diff
@@ -0,0 +1,58 @@
+import django_filters
+from .models import Restaurant
+
+class RestaurantFilter(django_filters.FilterSet):
+    title = django_filters.CharFilter(lookup_expr='icontains')
+    cost_order = django_filters.ChoiceFilter(
+        choices=[
+            ('', 'Sort by cost'),
+            ('low_to_high', 'Cost: Low to High'),
+            ('high_to_low', 'Cost: High to Low'),
+        ],
+        method='filter_by_cost',
+        empty_label=None
+    )
+    rating_order = django_filters.ChoiceFilter(
+        choices=[
+            ('', 'Sort by rating'),
+            ('low_to_high', 'Rating: Low to High'),
+            ('high_to_low', 'Rating: High to Low'),
+        ],
+        method='filter_by_rating',
+        empty_label=None
+    )
+    dietary_preference = django_filters.ChoiceFilter(
+        choices=Restaurant.DIETARY_CHOICES,
+        empty_label='All dietary types',
+        method='filter_by_dietary_preference'
+    )
+    spotlight = django_filters.BooleanFilter(widget=django_filters.widgets.BooleanWidget(), method='filter_by_spotlight')
```

Comment:
I don't think you need a separate method to filter_by_spotlight(), I think it will automatically handle.